### ISSUE TITLE
ExecStartPost for systemd-nspawn@ service broken in v239 with unified cgroup hierarchy

### ISSUE THREAD

**Issue by arianvp** [systemd/systemd#10482]
**systemd version the issue has been seen with**
> v239

**Used distribution**
 > NixOS 18.09
Distribution issue tracker for this issue: https://github.com/NixOS/nixpkgs/issues/48749

**Expected behaviour you didn't see**
In NixOS we have an `ExecStartPost` and an `ExecStopPre` script that set up
and tear down the network for our `systemd-nspawn` containers that are  running on the OS.
We have been using this for quite some time, and it used to work with Cgroup v2 in NixOS `18.03`
(See https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/virtualisation/containers.nix if you're interested in our upstream unit files.   We renamed `systemd-nspawn@` to `container@`).

**Unexpected behaviour you saw**
The container fails to start, with a scary CGROUP error, and `systemctl start` never terminates
```
Oct 21 22:07:11 ryzen systemd[25618]: systemd-nspawn@jessie.service: Failed to attach to cgroup /machine.slice/systemd-nspawn@jessie.service: Device or resource busy
Oct 21 22:07:11 ryzen systemd[25618]: systemd-nspawn@jessie.service: Failed at step CGROUP spawning /usr/bom/echo: Device or resource busy
```

**Steps to reproduce the problem**

1. Enable `systemd.unified_cgroup_hierarchy=1` in your kernel commandline

2.  Boostrap a jessie image 
    ```
    # debootstrap jessie /var/lib/machines/jessie
    ```

3.  Add an `ExecStartPost` or `ExecStopPre` script for the container:
   ```
   #   /etc/systemd/system/systemd-nspawn@.d/override.conf:
   # OR
   # /etc/systemd/system/systemd-nspawn@jessie.service.d/override.conf:
   [Service]
   ExecStartPost=/usr/bin/echo hello
   # OR: ExecStopPre=/usr/bin/echo hello
   ```

4.  `systemctl start systemd-nspawn@jessie`
4.   The `ExecStart` script completes, and boots the container, but `ExecStartPost/ExecStopPre` script crashes with the error:
```
Oct 21 22:07:11 ryzen systemd[25618]: systemd-nspawn@jessie.service: Failed to attach to cgroup /machine.slice/systemd-nspawn@jessie.service: Device or resource busy
Oct 21 22:07:11 ryzen systemd[25618]: systemd-nspawn@jessie.service: Failed at step CGROUP spawning /usr/bin/echo: Device or resource busy
```

5.  usually `systemctl start` _forever_ will hang when this occurs

**Comment by poettering** [comment]
hmm, yeah, this is caused by the no-process-in-inner-nodes rule of cgroupsv2. nspawn moves itself and the container payload down into subcgroups and this means we cannot start ExecStartPost= anymore in the cgroup itself. I figure we need to tweak this and always run ExecStartPost= in a subcgroup if Delegate=yes is set.

**Comment by poettering** [comment]
fix waiting in #10813
